### PR TITLE
Fixes #244. Skip ignored modules when calling time

### LINKED ISSUE #244: Why doesn't `ignore` work for email.utils.formatdate?
Example:

```python
    def test_foo(self):
        from email.utils import formatdate
        import ipdb
        ipdb.set_trace()
        print(formatdate())
        with freeze_time('2012-10-15', ignore=['email']):
            print(formatdate())
        assert 0
```

What's the expected behavior of the above usage of `ignore`? I'm expecting that it will not replace `time.time` inside `formatdate`, yet that appears to be what happens:

```python
> test.py(509)test_foo()
    508         ipdb.set_trace()
--> 509         print(formatdate())
    510         with freeze_time('2012-10-15', ignore=['email']):

ipdb> s
--Call--
> /usr/lib/python2.7/email/utils.py(124)formatdate()
    123 
--> 124 def formatdate(timeval=None, localtime=False, usegmt=False):
    125     """Returns a date string as specified by RFC 2822, e.g.:

ipdb> n
> /usr/lib/python2.7/email/utils.py(142)formatdate()
    141     # 2822 requires that day and month names be the English abbreviations.
--> 142     if timeval is None:
    143         timeval = time.time()

ipdb> n
> /usr/lib/python2.7/email/utils.py(143)formatdate()
    142     if timeval is None:
--> 143         timeval = time.time()
    144     if localtime:

ipdb> b 143
Breakpoint 1 at /usr/lib/python2.7/email/utils.py:143
ipdb> print time.time
<built-in function time>
ipdb> print time.time()
1527821322.89
ipdb> c
Fri, 01 Jun 2018 02:48:46 -0000
> /usr/lib/python2.7/email/utils.py(143)formatdate()
    142     if timeval is None:
1-> 143         timeval = time.time()
    144     if localtime:

ipdb> print time.time
<freezegun.api.FakeTime object at 0x7f1e2e624890> 
ipdb> print time.time()
1350259200.0
ipdb> c
Mon, 15 Oct 2012 00:00:00 -0000
```

### PR DESCRIPTION
**Pull request by kylegibson**
This is a proof of concept. If this solution is acceptable, I can extend the implementation to the other functions.

### CODE REVIEW

File: freezegun/api.py [hunk 1]
```diff
@@ -125,13 +126,36 @@ def with_metaclass(meta, *bases):
 )
 
 
-class FakeTime(object):
+class BaseFakeTime(object):
```
[the remark this answers is not in the document]
Comment (kylegibson) [reply freezegun/api.py]: Yes, because this logic will need to be applied to the other Fake* classes.

[the remark this answers is not in the document]
Comment (spulec) [reply freezegun/api.py]: Ah, got it. Should we do that now or will it require a lot more work? I feel a bit weird merging the functionality that will only work on some of the mocking.

[the remark this answers is not in the document]
Comment (kylegibson) [reply freezegun/api.py]: I will do that now, for this PR, if you give the +1.

[the remark this answers is not in the document]
Comment (spulec) [reply freezegun/api.py]: Sounds great!

[the remark this answers is not in the document]
Comment (spulec) [reply freezegun/api.py]: If you wouldn't mind adding a test or two, that would also be great. I don't want us to accidentally regress later.

**Comment by kylegibson** [comment]
@spulec Any thoughts on this PR?

**Comment by spulec** [comment]
Looks great; thanks for all your work!

**Pull request merged**